src/trace_processor/containers/interval_tree.h

```diff
@@ -39,6 +39,23 @@ struct Interval {
   Id id;
 };
 
+inline bool IsOverlapping(bool query_is_instant,
+                          Ts s,
+                          Ts e,
+                          const Interval& i) {
+  bool interval_is_instant = (i.start == i.end);
+  if (query_is_instant && interval_is_instant) {
```

Comment:
I believe instants at both the start and end of the interval should be passed through.